eureka_ml_insights/data_utils/spatial_utils.py

```diff
@@ -198,181 +204,65 @@ def extract_answer_from_text_map_and_maze(model_output_raw, options):
 
     # get dict of options from options string
     options_dict = {x.split(".")[0].strip().lower():x.split(".")[1].strip().lower() for x in options}
-
-
-    model_output_parsed_letter = ""
-    model_output_parsed = ""
             
     answers = [v for k, v in options_dict.items()]
     answers_pattern = rf"\b({'|'.join(answers)})\b"
 
-    if "Answer:".lower() in model_output_raw.lower():
-        pattern_letter = r"^\**Answer:\**\s+(\w)\. (\w+)"
-        matches = re.search(pattern_letter, model_output_raw, re.IGNORECASE)
+    parsed_answers = []
+
+    # Search over two different typical patterns for reporting the answer
+    search_patterns = [r"^\**Answer:\**\s+(\w)\. (\w+)", r'answer is:*\s*\**([\w\d]+)[\s:.]*\**'] 
+
+    for pattern in search_patterns:
+
+        # search for a single letter or digit answer
+        matches = re.findall(pattern, model_output_raw, re.IGNORECASE)
         if matches:
-            match_option = matches.group(1).lower()
-            if match_option in options_dict:
-                model_output_parsed_letter = options_dict[match_option]
-            else:
-                model_output_parsed_letter = match_option
+            match_option = matches[match_index]
+            if match_option in options_dict: #check if it is a letter, and then return the associated answer value
+                parsed_answer_value = options_dict[match_option]
+            else: #otherwise assume it is an answer value
+                parsed_answer_value = match_option
+            
+            parsed_answers = check_and_add_answer(parsed_answer_value, answers, parsed_answers)
 
+        #first grab the phrase after the "Answer:" tag then search in that phrase for the which of the options are present
         pattern_phrase = r"Answer:\**\s+([^\n]+)"
```

Comment:
How come you still have this pattern here even though you are iterating over patterns in the for loop?!